src/components/SideNav/Provider.js

```diff
@@ -1,58 +1,54 @@
 import React from "react";
 import PropTypes from "prop-types";
+import throttle from "lodash/throttle";
 
 import { DOM_TARGETS } from "constants/domNodes";
 
 import { findActiveNode } from "helpers/dom";
 
 const sortByPosition = (a, b) => {
-  const aY = a.current.getBoundingClientRect().top;
-  const bY = b.current.getBoundingClientRect().top;
+  const aY = a.ref.current.getBoundingClientRect().top;
+  const bY = b.ref.current.getBoundingClientRect().top;
   return aY - bY;
 };
 let lastScrollPosition = 0;
 
 export const SideNavProgressContext = React.createContext({});
 
 export const Provider = ({ children }) => {
-  const [activeNode, setActiveNode] = React.useState();
+  const [activeContent, setActiveNode] = React.useState({ id: "", ref: null });
   const [trackedElements, setTrackedElements] = React.useState([]);
 
-  // We need to search backwards pretty frequently, so memoize it so we don't
-  // generate a ton of garbage.
-  const backwardsElements = React.useMemo(
-    () => trackedElements.slice().reverse(),
-    [trackedElements],
-  );
+  // We need to search just the refs pretty frequently, so memoize it so we
+  // don't generate a ton of garbage.
+  const elementRefs = React.useMemo(() => trackedElements.map((e) => e.ref), [
+    trackedElements,
+  ]);
 
   React.useEffect(() => {
     const contentDom = document.querySelector(`#${DOM_TARGETS.contentColumn}`);
 
-    const handler = () => {
-      // If we haven't scrolled at least 20 pixels, just bail.
-      if (Math.abs(contentDom.scrollY - lastScrollPosition) < 20) {
+    const handler = throttle(() => {
+      // If we haven't scrolled at least 100 pixels, just bail.
+      if (Math.abs(contentDom.scrollY - lastScrollPosition) < 100) {
```

Comment:
super small: this 100 "threshold" is used in scrollrouter, too - might be nice to save it as a const somewhere

Reply:
Yeah there's a lot of room to unify the behavior there, I think, but it started to look like a big effort with relatively little benefit to the site. Maybe one day.